src-v2/Judge/NTS.Judge.Blazor/Pages/Dashboard/Components/ActionsPanelComponent.razor

```diff
@@ -0,0 +1,334 @@
+    </MudPaper>
+    <MudPaper Class="d-flex flex-row mb-4" Elevation="0">
+        <MudButton Class="@ReinspectionClasses" Variant="Variant.Filled" 
+                StartIcon="@Icons.Material.Filled._360" Color="Color.Error" 
+                @ref="reinspectionButton" OnClick="ToggleReinspection">
+            Reinspection
+        </MudButton>
+        <MudButton Class="@ReqInsClasses" Variant="Variant.Filled"
+                   StartIcon="@Icons.Material.Filled.HourglassEmpty" Color="Color.Primary"
+                   @ref="reqInsButton" OnClick="ToggleReqIns">
+            Required Inspection
+        </MudButton>
+    </MudPaper>
+    <MudPaper Class="d-flex flex-column mb-2" Elevation="0">
+        <MudButtonGroup Color="Color.Secondary" Size="Size.Medium" Variant="Variant.Outlined">
+            <MudButton Class="@WitClasses" OnClick="ToggleWitdrawn">Witdrawn</MudButton>
+            <MudButton Class="@RetClasses" OnClick="ToggleRetired">Retired</MudButton>
+            <MudButton Class="@FNRClasses" OnClick="ToggleFNR">Finished Not Ranked</MudButton>
+            <MudButton Class="@DQClasses" OnClick="ToggleDQ">Disqualified</MudButton>
+            <MudButton Class="@FTQClasses" OnClick="ToggleFTQ">Failed to Qualify</MudButton>
+        </MudButtonGroup>
+        <MudSelect @ref="multiSelect" T="string" Class="@MultiSelectClasses" Label="Failed to Qualify Reason" MultiSelection="true" Delimiter="+" @bind-Value="@FTQReasonValue" Placeholder="Select a reason">
+            @foreach (var failToQualifyCode in MultiSelectListModel.MultiSelectList<FTQCodes>())
+            {
+                <MudSelectItem T="string" Value="@failToQualifyCode.Value">
+                    @failToQualifyCode.Value
+                    <div class="font-size-10">@failToQualifyCode.Description</div>
+                </MudSelectItem>
+            }
+        </MudSelect>
+        <MudTextField @ref="reasonInput" @bind-Value="Reason" Class="@ReasonClasses" Label="Reason" Variant="Variant.Outlined"></MudTextField>
+    </MudPaper>
+
+    <MudButton Class="mt-4 one-third-width" Variant="Variant.Filled" Color="Color.Primary" OnClick="Update">
+        @Localizer.Get("Update")
+    </MudButton>
+</MudPaper>
+
+@code {
+    [Inject]
+    IParticipationBehind _participationBehind { get; set; } = default!;
+    [Parameter]
+    public int TandemNumber { get; set; }
+    private MudTextField<string> timeInput = default!;
+    private MudBaseButton reinspectionButton = default!;
+    private MudButton reqInsButton = default!;
+    private MudTextField<string> reasonInput = default!;
+    private MudSelect<string> multiSelect = default!;
+    private Participation? _selectedParticipation = default!;
+    private Phase? _currentPhase = default!;
+    private Snapshot _snapshot = default!;
+    private string Time { get; set; } = default!;
```

Comment:
Time has to be nullable. You should be able to update without setting Arrive or Inspection times.